### PR TITLE
Add bilateral filter

### LINKED ISSUE #702: [Feat] Bilateral filter
## 🚀 Feature

Would be great to have a differentiable version of Bilater Filtering `kornia.filter.bilateral_blur`

We can follow opencv signature:
https://docs.opencv.org/3.4/d4/d86/group__imgproc__filter.html#ga9d7064d478c95d60003cf839430737ed

Resource: http://jamesgregson.ca/bilateral-filtering-in-python.html

### PR DESCRIPTION
**Pull request by gau-nernst**
#### Changes

Closes #702 

Implement bilateral. Function signature follows OpenCV.

Regarding `color_distance_type`, use L1 to match OpenCV. Use L2 to match Matlab.

TODO

- [x] Add documentation
- [x] Add tests to compare against OpenCV (grayscale and RGB images)
- [x] Add ability to use batch of `sigma_color` and `sigma_distance`

@edgarriba @ducha-aiki Let me know your thoughts and feedback.

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Review by ducha-aiki — APPROVED** [review]
LGTM

File: kornia/filters/bilateral.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    border_type: str = 'reflect',
+    color_distance_type: str = "l1",
+) -> Tensor:
+    KORNIA_CHECK_IS_TENSOR(input)
+    KORNIA_CHECK_SHAPE(input, ['B', 'C', 'H', 'W'])
+
+    kx, ky = _unpack_2d_ks(kernel_size)
+    pad_x, pad_y = _compute_zero_padding(kernel_size)
+
+    padded = F.pad(input, (pad_x, pad_x, pad_y, pad_y), mode=border_type)
```
Comment (edgarriba) [inline kornia/filters/bilateral.py:27]: we have exposed `pad` in core as a common used ops
https://github.com/kornia/kornia/blob/master/kornia/core/__init__.py#L13

**Review by edgarriba — APPROVED** [review]
LGTM. Amazing! good as soon tests pass

File: kornia/filters/bilateral.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+    Examples:
+        >>> input = torch.rand(2, 4, 5, 5)
+        >>> output = bilateral_blur(input, (3, 3), 0.1, (1.5, 1.5))
+        >>> output.shape
+        torch.Size([2, 4, 5, 5])
+    """
+    KORNIA_CHECK_IS_TENSOR(input)
+    KORNIA_CHECK_SHAPE(input, ['B', 'C', 'H', 'W'])
+    if isinstance(sigma_color, Tensor):
```
Comment (johnnv1) [inline kornia/filters/bilateral.py:53]: in theory, with the last updates on kornia check, this should work
```suggestion
    if KORNIA_CHECK_IS_TENSOR(sigma):
```

Comment (johnnv1) [reply kornia/filters/bilateral.py:53]: oh, true, 😅

**Pull request merged**